test.sh

```diff
@@ -0,0 +1,251 @@
+        echo 'Error! The Triton CLI configuration does not match the Docker CLI configuration.'
+        tput sgr0 # clear
+        echo
+        echo "Docker user: ${docker_user}"
+        echo "Triton user: ${TRITON_USER}"
+        echo "Docker data center: ${docker_dc}"
+        echo "Triton data center: ${TRITON_DC}"
+        exit 1
+    fi
+
+    local triton_cns_enabled=$(triton account get | awk -F": " '/cns/{print $2}')
+    if [ ! "true" == "$triton_cns_enabled" ]; then
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! Triton CNS is required and not enabled.'
+        tput sgr0 # clear
+        echo
+        exit 1
+    fi
+
+    echo CONSUL=consul.svc.${TRITON_ACCOUNT}.${TRITON_DC}.cns.joyent.com > _env
+}
+
+# ---------------------------------------------------
+# utility functions
+
+# get the address for a container via either local docker-machine or Triton CLI.
+# requires service name (or name + port + protocol for local use)
+getPublicUrl() {
+    if [ -z "${COMPOSE_FILE}" ]; then
+        local cname=$(printf '%s.svc.%s.%s.triton.zone' $1 ${TRITON_ACCOUNT} ${TRITON_DC})
+        echo "$cname:$port"
+    else
+        local name=$(echo $1 | sed 's/_/-/')
+        local ip=$(docker-machine ip default)
+        local port=$(getPort $name $2 $3)
+        echo "$ip:$port"
+    fi
+}
+
+# get the IP:port of a container's private IP via `docker exec`
+getPrivateIpPort() {
+    local ip=$(docker exec -it ${COMPOSE_PROJECT_NAME}_$1_1 ip addr show eth0 | grep -o '[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}')
+    local port=$(getPort $1 $2 $3)
+    echo "$ip:$port"
+}
+
+# get the mapped port number for a given container's port and protocol
+getPort() {
+    local protocol=$3
+    if [ -z $protocol ]; then
+        protocol='tcp'
+    fi
+    if [ -z "${COMPOSE_FILE}" ]; then
+        local port=$2
+    else
+        local port=$(docker inspect ${COMPOSE_PROJECT_NAME}_$1_1 | json -a NetworkSettings.Ports."$2/$protocol" | json -a HostPort | sort -nb | head -1)
+    fi
+    echo $port
+}
+
+# usage: poll-for-page <url> <pre-message> <post-message>
+poll-for-page() {
+    echo "$2"
+    while :
+    do
+        curl --fail -s -o /dev/null "$1" && break
+        sleep 1
+        echo -ne .
+    done
+    echo
+    echo "$3"
+    open "$1"
+}
+
+doStuff() {
+    echo doStuff
+}
+
+# ---------------------------------------------------
+# parse arguments
+
+while getopts "f:p:h" optchar; do
+    case "${optchar}" in
+        f) export COMPOSE_FILE=${OPTARG} ;;
+        p) export COMPOSE_PROJECT_NAME=${OPTARG} ;;
+    esac
+done
+shift $(expr $OPTIND - 1 )
+
+until
+    cmd=$1
+    if [ ! -z "$cmd" ]; then
+        shift 1
+        $cmd "$@"
+        if [ $? == 127 ]; then
+            help
+        fi
+        exit
+    fi
+do
+    echo
+done
+
+# default behavior
+echo "Starting example application"
+echo "project prefix:      $COMPOSE_PROJECT_NAME"
+echo "docker-compose file: $COMPOSE_FILE"
```

Comment:
Yeah, I'm feeling we should either emit the file we're using here or conditionally not emit the static text label.